### ISSUE TITLE
MachineConfigPool KeyError

### ISSUE THREAD

**Issue by bostrt** [kxr/o-must-gather#14]
Failure parsing a must-gather from 4.5.15 cluster. 

```
$ pip3 show o-must-gather
Name: o-must-gather
Version: 1.0.0
...

$ omg get mcp 
Traceback (most recent call last):
  File "/home/rdu/rbost/.local/bin/omg", line 11, in <module>
    load_entry_point('o-must-gather==1.0.0', 'console_scripts', 'omg')()
  File "/home/rdu/rbost/.local/lib/python3.6/site-packages/omg/cli.py", line 77, in main
    func(args)
  File "/home/rdu/rbost/.local/lib/python3.6/site-packages/omg/cmd/get_main.py", line 150, in get_main
    getout_func(rt, ns, res, a.output, show_type)
  File "/home/rdu/rbost/.local/lib/python3.6/site-packages/omg/cmd/get/mcp_out.py", line 20, in mcp_out
    row.append(mcp['spec']['configuration']['name'])
KeyError: 'configuration'
```

Here's the spec from the Infra MCP. There's no machines in the MCP which leads to the empty `configuration` block:

```
spec:
  machineConfigSelector:
    matchExpressions:
    - key: machineconfiguration.openshift.io/role
      operator: In
      values:
      - infra
  nodeSelector:
    matchLabels:
      node-role.kubernetes.io/infra: ""
```

**Comment by kxr** [comment]
Handling of missing mcp['spec']['configuration'] fixed in commit [51da0e4](https://github.com/kxr/o-must-gather/commit/51da0e42bf8b4728f9e5c96a8f1a7d658f38996b) (included in o-must-gather-1.0.1).

Can you test it out. I am suspecting that in this case other fields in mcp['status'] might also be missing.

**Comment by kxr** [comment]
@bostrt Handling of missing mcp['spec']['configuration'] fixed in commit [51da0e4](https://github.com/kxr/o-must-gather/commit/51da0e42bf8b4728f9e5c96a8f1a7d658f38996b) (included in o-must-gather-1.0.1).

Can you test it out. I am suspecting that in this case other fields in mcp['status'] might also be missing.

**Comment by bostrt** [comment]
Thanks!! It looks good now. Using the same must-gather causing issues before:

```
$ omg version
omg version 1.0.1 (https://github.com/kxr/o-must-gather)

$ omg get mcp 
NAME    CONFIG                                            UPDATED  UPDATING  DEGRADED  MACHINECOUNT  READYMACHINECOUNT  UPDATEDMACHINECOUNT  DEGRADEDMACHINECOUNT  AGE
infra                                                                                  0             0                  0                    0                     10d
master  rendered-master-60c334e3b4906bda11c1dbd91cba8263  False    True      True      3             2                  2                    1                     10d
router                                                                                 0             0                  0                    0                     10d
worker  rendered-worker-dc7aedafc57aab543f5ba91dfcc4823d  False    True      False     4             3                  4                    0                     10d
```
